html/admin/assets/widgets/calendar.twig

```diff
@@ -91,47 +62,23 @@
-                                    {% set earliestStart = subproject.projects_dates_use_start|date("c") %}
-                                {% endif %}
-                                {% if latestEnd < subproject.projects_dates_use_end|date("c") or latestEnd == false %}
-                                    {% set latestEnd = subproject.projects_dates_use_end|date("c") %}
-                                {% endif %}
                             {% endif %}
                         {% endfor %}
                     {% endfor %}
                 ],
-                validRange: {
-                    start: '{{ earliestStart|date("c") }}',
-                    end: '{{ latestEnd|date("c") }}'
-                },
-                editable  : false,
-                droppable : false, // this allows things to be dropped onto the calendar !!!
-                drop      : function(info) {
-
-                },
-                height: 800,
-                firstDay: 1,
                 weekNumbers: true,
-                weekNumbersWithinDays:true,
-                {% if USERDATA.instance.weekStartDates %}
-                weekNumberCalculation: function(date) {
-                    var weekStartDates = [{{ USERDATA.instance.weekStartDates|join(",") }}];
-                    weekStartDates.push(32472144000000); // add a date to the end of the list as otherwise the last one is ignored
-                    var today = new Date();
-                    var dateBeingTested = Date.parse(date.getDate() + ' ' + (date.toLocaleString('default', { month: 'long' })) + ' ' + date.getFullYear() + ' 00:00:00');
-                    var idOfThis;
-                    for (var i = 0; i < weekStartDates.length; i++) {
-                        if (dateBeingTested <= (weekStartDates[i])-(7 * 24 * 60 * 60 * 1000)) { // If this date is before the week start date, then it is in the previous week.
-                            if (i == 0) {
-                                return null;
-                            }
-                            idOfThis = i-1;
-                            break;
+                {% if USERDATA.instance['instances_weekStartDates'] %}
+                weekNumberContent : function(week) {
+                    var weekStartDates =  {{USERDATA.instance['instances_weekStartDates']|raw}};
+                    for (const [date, label] of Object.entries(weekStartDates)) {
+                        if (moment(week.date).isSame(date, 'isoWeek')) {
+                            //This date is in this week so return the label
+                            return (label);
                         }
                     }
-                    if (typeof idOfThis === 'undefined') return null;
-                    return (Math.round((dateBeingTested-weekStartDates[idOfThis]) / (7 * 24 * 60 * 60 * 1000))+1);
+                    //default to week number if it doesn't have a label
```

Comment:
I can't decide if it's better to have it blank rather than have the week number, is is confusing to have random numbers involved as well?

Reply:
Added a default of a space character which blanks it